src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -425,32 +431,47 @@ private enum AssignmentQueryKind {
 
 					return AssignmentQueryKind.Hopeless;
 
-				default:
-					var model = m_compilation.GetSemanticModel( assignment.Expression.SyntaxTree );
-					var typeInfo = model.GetTypeInfo( assignment.Expression );
-
-					// Type can be null in the case of an implicit conversion where the
-					// expression alone doesn't have a type. For example:
-					//   int[] foo = { 1, 2, 3 };
-					var typeToCheck = typeInfo.Type ?? typeInfo.ConvertedType;
-
-					if( assignment.Expression is BaseObjectCreationExpressionSyntax _ ) {
-						// When we have a new T() we don't need to worry about the value
-						// being anything other than an instance of T.
-						query = new ImmutabilityQuery(
-							ImmutableTypeKind.Instance,
-							typeToCheck
-						);
-					} else {
-						// In general we need to handle subtypes.
-						query = new ImmutabilityQuery(
-							ImmutableTypeKind.Total,
-							typeToCheck
-						);
+				case SyntaxKind.InvocationExpression:
+					var semanticModel = m_compilation.GetSemanticModel( assignment.Expression.SyntaxTree );
+					var symbol = semanticModel.GetSymbolInfo( assignment.Expression ).Symbol;
+
+					if( symbol is not IMethodSymbol methodSymbol ) {
+						break;
 					}
 
-					return AssignmentQueryKind.ImmutabilityQuery;
+					var symbolName = methodSymbol.ContainingSymbol.ToDisplayString() + "." + methodSymbol.Name;

```

Comment:
Not super hot on the string comparisons, though I'm struggling to find a way to do it otherwise.